### ISSUE TITLE
SI4T "Generate Index Data" TBB does not render component presentations which are in a region

### ISSUE THREAD

**Issue by jhorsman** [SI4T/SI4T#20]
Sites 9 introduced the concept of page regions in the Sites Content Manager. Component presentations can be organized in regions on a page.

The "Generate Index Data" TBB ignores those regions and only renders component presentations which are added directly on the page.

**Comment by willprice76** [comment]
Hi Jan.

If you want to fix the code I can make you a contributor. Shouldnt be too
complicated. Would be more interesting to look at excluding particular
regions from indexing... do they have metadata?

Thanks
W.

**Comment by neeteshnarvaria** [comment]
Hi @willprice76 ,

I have done code changes on my local to generate index data for both page component presentation and region component presentation as suggested by @jhorsman . i have created separate branch.

Please provide me access to push the changes.

Thanks,
Neetesh

**Comment by willprice76** [comment]
Thanks for making the fix. I Added you to the developers group. Please make
a pull request. You can either merge it yourself or ask one of the other
devs to review and merge

**Comment by avmgan** [comment]
Hi @willprice76,

I'm starting testing this fix and review the code for the merge, will keep you posted.

Thanks,
Velu

**Comment by jhorsman** [comment]
@willprice76 
> Would be more interesting to look at excluding particular regions from indexing... do they have metadata?
Region schemas can have metadata fields indeed, this meta data is set in the page item where the region is applied. There is no meta data on the region schema itself, just like content schemas do not have meta data applied to the schema itself.

I am not sure if there is a use case where an editor would like to configure for each page if specific regions are indexed or not.

**Comment by willprice76** [comment]
True. I actually meant that the region would always be excluded, not per
page.
